### ISSUE TITLE
Few weapons don't have additional effects - trying to write a script for them

### ISSUE THREAD

**Issue by personaone** [DerpyProjectGroup/topaz#511]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened
- [x] I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I have specified what branch this happens on branch: Base

The spear Dabo/Dabo+1 doesn't have an additional Lightning damage effect scripted it looks like, also a few Thief daggers like the Palladium, I'd happily write a script but I can't seem to get it working, I used the Sirocco Kukri one as an example and tried to edit it in order for Thunder/Lightning replace the Wind damage, but it doesn't want to work for some reason. Here's a sample:

-----------------------------------
-- ID: 18130
-- Item: Dabo +1
-- Additional Effect: Lightning damage
-----------------------------------
require("scripts/globals/status")
require("scripts/globals/magic")
require("scripts/globals/msg")
-----------------------------------
local item_object = {}

item_object.onAdditionalEffect = function(player, target, damage)
    local dmg = math.random(3, 10)
    local params = {}
    params.bonusmab = 0
    params.includemab = false
    dmg = addBonusesAbility(player, xi.magic.ele.THUNDER, target, dmg, params)
    dmg = dmg * applyResistanceAddEffect(player, target, xi.magic.ele.THUNDER, 0)
    dmg = adjustForTarget(target, dmg, xi.magic.ele.THUNDER)
    dmg = finalMagicNonSpellAdjustments(player, target, xi.magic.ele.THUNDER, dmg)

    local message = xi.msg.basic.ADD_EFFECT_DMG
    if (dmg < 0) then
        message = xi.msg.basic.ADD_EFFECT_HEAL
    end

    return xi.subEffect.LIGHTNING_DAMAGE, message, dmg
end

return item_object

Not really familiar with how this thing works but I'm willing to spend the time if anyone can help out?

Also it appears a few rings are affected, so far I saw Hercules' Ring not proccing when HP is lower than 50% as it should, so let me sum up and keep adding stuff that I test that doesn't work:
Dabo, Dabo+1, Palladium Dagger, Bartholomew's Knife, Hercules' Ring

**Comment by RAIST5150** [comment]
Been a long standing issue... elemental staves also don't proc their elemental damage.

**Comment by personaone** [comment]
oops - I didn't see any issue related to that in the tracker sorry

**Comment by KnowOne134** [comment]
You have to add the mod 431 to item_mods.sql to trigger the script it has an additional script

**Comment by personaone** [comment]
> You have to add the mod 431 to item_mods.sql to trigger the script it has an additional script

Ah - got it, will work on it, got a few scripts I want to contribute if possible

**Comment by KnowOne134** [comment]
I have a global file use and an example I did for my server. But I believe @TeoTwawki was building something also to make this outdated

https://github.com/KnowOne134/DSP-Shared_Collection/tree/main/Weapon%20Additional%20Effects

**Comment by TeoTwawki** [comment]
These aren't being added anymore due to my refactor which is held up by having to keep rebasing it and then getting fatigued and not wanting to spend all of what little freetime I have just getting it back to functioning. The branch I had was working but needed tedious sql rows added, so it sat in a branch till stale (again). To fix some edge case I had to modify mob families, which is half done right now and after its completed I'll begin my rebase from hell to get to where I can finish it off. 

[This was where I left off, as you can see its been awhile and I haven't brought it to LandSandBoat yet..](https://github.com/topaz-next/topaz/tree/wip/effect-rework)

**Comment by personaone** [comment]
I see, should I even bother writing some of the scripts? I can do a few weapons that I know are missing the effects but since I haven’t validated the damage on retail I can only rely on wikis and common sense.

**Comment by TeoTwawki** [comment]
> I see, should I even bother writing some of the scripts? I can do a few weapons that I know are missing the effects but since I haven’t validated the damage on retail I can only rely on wikis and common sense.

Nope, would not bother. If there is a particular item you just gotta play with maybe, but better to wait or help on the newer system instead. I hope to have it at least in an lsb feature branch real soon

**Comment by personaone** [comment]
> > I see, should I even bother writing some of the scripts? I can do a few weapons that I know are missing the effects but since I haven’t validated the damage on retail I can only rely on wikis and common sense.
> 
> Nope, would not bother. If there is a particular item you just gotta play with maybe, but better to wait or help on the newer system instead. I hope to have it at least in an lsb feature branch real soon

Understood. I can close this issue now or leave it open as I don’t know if it’s being tracked, just let me know.

**Comment by TeoTwawki** [comment]
prolly should leave open for now, as I don't think the issue got remade yet since topaz's last death and rebirth..

lots and lots of old "known" issues have lost tracking thanks to other peoples drama..

**Comment by personaone** [comment]
> prolly should leave open for now, as I don't think the issue got remade yet since topaz's last death and rebirth..
> 
> lots and lots of old "known" issues have lost tracking thanks to other peoples drama..

You got it :) shout if you need any help !
